src/mame/drivers/atarisy1.cpp

```diff
@@ -240,6 +239,13 @@ MACHINE_RESET_MEMBER(atarisy1_state,atarisy1)
 }
 
 
+WRITE_LINE_MEMBER(atarisy1_state::music_reset_w)
+{
+	// reset the YM2151 and YM3012
+	if (!state)
+		m_ymsnd->reset();
```

Comment:
"YM devices" is a fairly vast category; I'll remark that MAME's OPN emulation is a big ugly heap of legacy C code whose modernization would be a fairly Herculean coding task. Speaking only of those at hand, as my comment states, there are actually two devices being reset here, even though the emulation reasonably treats them as a single unit.